test/service/test_response.py

```diff
@@ -3506,6 +3509,25 @@ def test_projects_response(self):
         }
         self.assertEqual({None: 2, 'Lung': 1, 'Retina': 1, 'Blood': 1}, terms)
 
+    def test_data_use_restriction(self):
+        """
+        Verify the data_use_restriction field
+        """
+        params = {
+            'catalog': self.catalog,
+            'sort': 'dataUseRestriction',
+            'filters': json.dumps({'dataUseRestriction': {'is': ['NRES']}})
+        }
+        for entity_type in ('bundles', 'files', 'projects', 'samples'):
+            url = self.base_url.set(path=('index', entity_type), args=params)
+            response = requests.get(url)
+            response.raise_for_status()
+            response_json = response.json()
+            self.assertGreater(len(response_json['hits']), 0)
+            for hit in response_json['hits']:
+                expected = 'NRES' if entity_type == 'projects' else ['NRES']
+                self.assertEqual(expected, one(hit['projects'])['dataUseRestriction'])
```

Comment:
I specifically requested that the term facet part of the response be verified as well

https://github.com/DataBiosphere/azul/pull/6207#discussion_r1746292060